libs/api/ebpf_api.cpp

```diff
@@ -273,10 +270,12 @@ _map_lookup_element(
 {
     ebpf_result_t result = EBPF_SUCCESS;
     try {
-        ebpf_protocol_buffer_t request_buffer(sizeof(_ebpf_operation_map_find_element_request) + key_size - 1);
-        ebpf_protocol_buffer_t reply_buffer(sizeof(_ebpf_operation_map_find_element_reply) + value_size - 1);
-        auto request = reinterpret_cast<_ebpf_operation_map_find_element_request*>(request_buffer.data());
-        auto reply = reinterpret_cast<_ebpf_operation_map_find_element_reply*>(reply_buffer.data());
+        ebpf_protocol_buffer_t request_buffer(
+            EBPF_OFFSET_OF(ebpf_operation_map_find_element_request_t, key) + key_size);
+        ebpf_protocol_buffer_t reply_buffer(
+            EBPF_OFFSET_OF(ebpf_operation_map_find_element_reply_t, value) + value_size);
```

Comment:
why not change the request reply data structures to store the key_size and value_sizes? That way ebpf_core.c will not have to estimate the key/value sizes.

Reply:
We can't trust the key/value size in the request. We verify that header.length  == input/output buffer length. We then derive the value for how large the key/value that the caller supplied is from the delta from header.length and the known fields.

All checks that provider key size and provided value size are correct is done in ebpf_map.c

Reply:
Agreed. You can resolve this.